pyproject.toml

```diff
@@ -0,0 +1,65 @@
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "amc2moodle"
+dynamic = ['version']
+authors = [
+    {name = "B. Nennig", email = "benoit.nennig@isae-supmeca.fr"},
+    {name = "L. Laurent", email = "luc.laurent@lecnam.net"}
+]
+description = "A tool to convert automultiplechoice quizz to moodle questions"
+readme = "README.md"
+classifiers=[
+    "Programming Language :: Python :: 3",
+    "License :: OSI Approved :: GNU General Public License v3 (GPLv3)",
+    "Operating System :: OS Independent",
+]
+license = {file = "LICENSE"}
+dependencies = [
+    'wand>=0.5.9',
+    'lxml>=3.5.0',
+    'pyparsing>=2.4.6',
+    'wheel>=0.29.0',
+    'markdown'
+]
+requires-python='>=3.8'
+
+[project.urls]
+Homepage = "https://github.com/nennigb/amc2moodle"
+Repository = "https://github.com/nennigb/amc2moodle"
+Issues = "https://github.com/nennigb/amc2moodle/issues"
+
+[project.scripts]
+amc2moodle = "amc2moodle.amc2moodle.bin.amc2moodle:run"
+moodle2amc = "amc2moodle.moodle2amc.bin.moodle2amc:run"
+
+[tool.hatch.version]
+path = "amc2moodle/_version.py"
+
+[tool.hatch.build]
+packages = ["amc2moodle"]
+# only-packages = true
+ignore-vcs = false
```

Comment:
to be checked

Reply:
done

Reply:
Run `hatch build` and `unzip -l dist/amc2moodle-2.11.0-py3-none-any.whl `

- with `ignore-vcs=false`:

```
Archive:  dist/amc2moodle-2.11.0-py3-none-any.whl
  Length      Date    Time    Name
---------  ---------- -----   ----
     1214  02-02-2020 00:00   amc2moodle/__init__.py
       23  02-02-2020 00:00   amc2moodle/_version.py
    14865  02-02-2020 00:00   amc2moodle/amc2moodle/README.md
        0  02-02-2020 00:00   amc2moodle/amc2moodle/__init__.py
    14237  02-02-2020 00:00   amc2moodle/amc2moodle/amc2moodle_class.py
     9030  02-02-2020 00:00   amc2moodle/amc2moodle/automultiplechoice.sty.ltxml
    43296  02-02-2020 00:00   amc2moodle/amc2moodle/convert.py
     2350  02-02-2020 00:00   amc2moodle/amc2moodle/fp.sty.ltxml
      370  02-02-2020 00:00   amc2moodle/amc2moodle/mhchem.sty.ltxml
     4864  02-02-2020 00:00   amc2moodle/amc2moodle/transform.xslt
    15439  02-02-2020 00:00   amc2moodle/amc2moodle/transform2html.xslt
      741  02-02-2020 00:00   amc2moodle/amc2moodle/transform_ns.xslt
     4581  02-02-2020 00:00   amc2moodle/amc2moodle/transform_qtype.xslt
     6893  02-02-2020 00:00   amc2moodle/amc2moodle/bin/amc2moodle.py
     3606  02-02-2020 00:00   amc2moodle/moodle2amc/README.md
      169  02-02-2020 00:00   amc2moodle/moodle2amc/__init__.py
    24322  02-02-2020 00:00   amc2moodle/moodle2amc/_questions.py
    13056  02-02-2020 00:00   amc2moodle/moodle2amc/_quiz.py
     5691  02-02-2020 00:00   amc2moodle/moodle2amc/html2tex.xslt
     2708  02-02-2020 00:00   amc2moodle/moodle2amc/struc2tex.xslt
     5905  02-02-2020 00:00   amc2moodle/moodle2amc/bin/moodle2amc.py
     1512  02-02-2020 00:00   amc2moodle/tests/__init__.py
    18807  02-02-2020 00:00   amc2moodle/tests/test_amc2moodle.py
     4200  02-02-2020 00:00   amc2moodle/tests/test_moodle2amc.py
     7532  02-02-2020 00:00   amc2moodle/tests/test_utils_calculatedParser.py
     2616  02-02-2020 00:00   amc2moodle/tests/test_utils_text.py
     1312  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/4.png
   187722  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/QCM.pdf
     2981  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/QCM.tex
    63905  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/QCM.xml
     2983  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/QCM_wo-tikz.tex
    59980  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/QCM_wo-tikz.xml
     3143  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/cleaning.tex
     2171  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/cleaning.xml
    10906  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/common-bank.tex
     2573  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/element.tex
      440  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/includestyles.tex
     4397  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/numerical.tex
     4392  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/numerical.xml
      484  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/tikz.tex
     3386  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/Figures/tinymonk.pdf
     3896  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/Figures/other/4r.png
    14813  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/Figures/other/schema_interpL.png
    48501  02-02-2020 00:00   amc2moodle/tests/payload_test_moodle2amc/automultiplechoice.sty
   141172  02-02-2020 00:00   amc2moodle/tests/payload_test_moodle2amc/moodle-bank-exemple.pdf
    10728  02-02-2020 00:00   amc2moodle/tests/payload_test_moodle2amc/moodle-bank-exemple.tex
    29495  02-02-2020 00:00   amc2moodle/tests/payload_test_moodle2amc/moodle-bank-exemple.xml
        0  02-02-2020 00:00   amc2moodle/utils/__init__.py
    18196  02-02-2020 00:00   amc2moodle/utils/calculatedParser.py
     6151  02-02-2020 00:00   amc2moodle/utils/customLogging.py
     5658  02-02-2020 00:00   amc2moodle/utils/flatex.py
     2891  02-02-2020 00:00   amc2moodle/utils/misc.py
     2656  02-02-2020 00:00   amc2moodle/utils/text.py
    50136  02-02-2020 00:00   amc2moodle-2.11.0.dist-info/METADATA
       87  02-02-2020 00:00   amc2moodle-2.11.0.dist-info/WHEEL
      126  02-02-2020 00:00   amc2moodle-2.11.0.dist-info/entry_points.txt
    35139  02-02-2020 00:00   amc2moodle-2.11.0.dist-info/licenses/LICENSE
     5719  02-02-2020 00:00   amc2moodle-2.11.0.dist-info/RECORD
---------                     -------
   934166                     58 files
```

- with `ignore-vcs=true`

```
Archive:  dist/amc2moodle-2.11.0-py3-none-any.whl
  Length      Date    Time    Name
---------  ---------- -----   ----
     1214  02-02-2020 00:00   amc2moodle/__init__.py
       23  02-02-2020 00:00   amc2moodle/_version.py
    14865  02-02-2020 00:00   amc2moodle/amc2moodle/README.md
        0  02-02-2020 00:00   amc2moodle/amc2moodle/__init__.py
    16347  02-02-2020 00:00   amc2moodle/amc2moodle/amc2moodle_class.latexml.log
    14237  02-02-2020 00:00   amc2moodle/amc2moodle/amc2moodle_class.py
     9030  02-02-2020 00:00   amc2moodle/amc2moodle/automultiplechoice.sty.ltxml
    43296  02-02-2020 00:00   amc2moodle/amc2moodle/convert.py
     2350  02-02-2020 00:00   amc2moodle/amc2moodle/fp.sty.ltxml
      370  02-02-2020 00:00   amc2moodle/amc2moodle/mhchem.sty.ltxml
     4864  02-02-2020 00:00   amc2moodle/amc2moodle/transform.xslt
    15439  02-02-2020 00:00   amc2moodle/amc2moodle/transform2html.xslt
      741  02-02-2020 00:00   amc2moodle/amc2moodle/transform_ns.xslt
     4581  02-02-2020 00:00   amc2moodle/amc2moodle/transform_qtype.xslt
     6893  02-02-2020 00:00   amc2moodle/amc2moodle/bin/amc2moodle.py
     3606  02-02-2020 00:00   amc2moodle/moodle2amc/README.md
      169  02-02-2020 00:00   amc2moodle/moodle2amc/__init__.py
    24322  02-02-2020 00:00   amc2moodle/moodle2amc/_questions.py
    13056  02-02-2020 00:00   amc2moodle/moodle2amc/_quiz.py
     5691  02-02-2020 00:00   amc2moodle/moodle2amc/html2tex.xslt
     2708  02-02-2020 00:00   amc2moodle/moodle2amc/struc2tex.xslt
     5905  02-02-2020 00:00   amc2moodle/moodle2amc/bin/moodle2amc.py
     1512  02-02-2020 00:00   amc2moodle/tests/__init__.py
    18807  02-02-2020 00:00   amc2moodle/tests/test_amc2moodle.py
     4200  02-02-2020 00:00   amc2moodle/tests/test_moodle2amc.py
     7532  02-02-2020 00:00   amc2moodle/tests/test_utils_calculatedParser.py
     2616  02-02-2020 00:00   amc2moodle/tests/test_utils_text.py
     1312  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/4.png
   187722  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/QCM.pdf
     2981  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/QCM.tex
    63905  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/QCM.xml
    14198  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/QCM_iptjftm5_magic.tex
     2983  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/QCM_wo-tikz.tex
    59980  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/QCM_wo-tikz.xml
    13737  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/QCM_wo-tikz_9tvop5bq_magic.tex
     3143  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/cleaning.tex
     2171  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/cleaning.xml
     3143  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/cleaning_8e6huv9h_magic.tex
    10906  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/common-bank.tex
     2573  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/element.tex
      440  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/includestyles.tex
     4397  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/numerical.tex
     4392  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/numerical.xml
      484  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/tikz.tex
     2370  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/xmlclean67042zf5
     3386  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/Figures/tinymonk.pdf
     3896  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/Figures/other/4r.png
    14813  02-02-2020 00:00   amc2moodle/tests/payload_test_amc2moodle/Figures/other/schema_interpL.png
    48501  02-02-2020 00:00   amc2moodle/tests/payload_test_moodle2amc/automultiplechoice.sty
   141172  02-02-2020 00:00   amc2moodle/tests/payload_test_moodle2amc/moodle-bank-exemple.pdf
    10728  02-02-2020 00:00   amc2moodle/tests/payload_test_moodle2amc/moodle-bank-exemple.tex
    29495  02-02-2020 00:00   amc2moodle/tests/payload_test_moodle2amc/moodle-bank-exemple.xml
        0  02-02-2020 00:00   amc2moodle/utils/__init__.py
    18196  02-02-2020 00:00   amc2moodle/utils/calculatedParser.py
     6151  02-02-2020 00:00   amc2moodle/utils/customLogging.py
     5658  02-02-2020 00:00   amc2moodle/utils/flatex.py
     2891  02-02-2020 00:00   amc2moodle/utils/misc.py
     2656  02-02-2020 00:00   amc2moodle/utils/text.py
    50136  02-02-2020 00:00   amc2moodle-2.11.0.dist-info/METADATA
       87  02-02-2020 00:00   amc2moodle-2.11.0.dist-info/WHEEL
      126  02-02-2020 00:00   amc2moodle-2.11.0.dist-info/entry_points.txt
    35139  02-02-2020 00:00   amc2moodle-2.11.0.dist-info/licenses/LICENSE
     6316  02-02-2020 00:00   amc2moodle-2.11.0.dist-info/RECORD
---------                     -------
   984558                     63 files
```

Reply:
Note in the second case, 3 addtional files that are not managed by VCS (`*_magic.tex` and `xmlcleanXXX`). So we can use `ignore-vcs=false`.

Reply:
yep, with a clean working environment :-)

Reply:
In my case, when I ran thes tests, my cwd was not clean at all but I obtain the previous results.